File: src/main/java/me/geek/mail/common/Menu/Sub/Micon.java
package me.geek.mail.common.Menu.Sub;

import java.util.List;

/**
 * 作者: 老廖
 * 时间: 2022/7/5
 **/
public final class Micon {
    private final String Icon;
    private final IconType Type;
    private final String Mats;
    private final int Data;
    private final String Name;
    private final List<String> Lore;

    public Micon(String icon, IconType type, String mats, int data, String name, List<String> lore) {
        this.Icon = icon;
        this.Type = type;
        this.Mats = mats;
        this.Data = data;
        this.Name = name;
        this.Lore = lore;
    }

    public String getIcon() {
        return Icon;
    }
    public IconType getType() {
        return this.Type;
    }

    public String getMats() {
        return Mats;
    }

    public int getData() {
        return Data;
    }

    public String getName() {
        return Name;
    }

    public List<String> getLore() {
        return Lore;
    }

}

File: src/main/kotlin/me/geek/mail/common/menu/sub/Micon.kt
package me.geek.mail.common.menu.sub

import me.geek.mail.utils.colorify
import taboolib.library.configuration.ConfigurationSection
import java.util.*


/**
 * 作者: 老廖
 * 时间: 2022/7/5
 */
class Micon(
    val icon: String,
    val type: IconType,
    val mats: String,
    val data: Int,
    val name: String,
    val lore: List<String>
) {
    constructor(icon: String, obj: ConfigurationSection) : this(
        icon,
        IconType.valueOf(obj.getString("Type", "NORMAL")!!.uppercase(Locale.ROOT)),
        obj.getString("display.mats","PAPER")!!,
        obj.getInt("display.data",0),
        obj.getString("display.name", " ")!!.colorify(),
        obj.getStringList("display.lore").joinToString().colorify().split(", "),
    )
}